src/material-examples/autocomplete-overview/autocomplete-overview-example.ts

```diff
@@ -72,9 +72,16 @@ export class AutocompleteOverviewExample {
         .map(name => this.filterStates(name));
   }
 
+  escapeSpecChars(val: string) {
+    return val ? val.replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, '\\$&')
+               : val;
+  }
+
   filterStates(val: string) {
-    return val ? this.states.filter(s => new RegExp(`^${val}`, 'gi').test(s))
+    let escVal: string = this.escapeSpecChars(val);
+    return val ? this.states.filter(s => new RegExp(`^${escVal}`, 'gi').test(s))
```

Comment:
I think a cleaner solution would be to just not use a regex at all and instead just use `indexOf`